src/Split.tsx

```diff
@@ -211,27 +211,32 @@ export class Split extends React.Component<SplitProps, ISplitState> {
     this.props.onDisable?.(disable);
   };
   setSize = (size: number, updateRatio?: boolean) => {
-    const sideSize = this.getContainerSize();
+    let sideSize = this.getContainerSize();
     if (sideSize === -1) {
       return;
     }
 
     let newSize = size;
     let mode = this.state.mode;
 
-    if (this.state.maxSize > -1 && newSize > this.state.maxSize) {
+    if (this.state.maxSize > -1 && newSize >= this.state.maxSize) {
       newSize = this.state.maxSize;
+      sideSize = sideSize - newSize;
+    } else if (this.state.maxSize < 0 && newSize >= sideSize + this.state.maxSize) {
+      newSize = sideSize + this.state.maxSize;
+      sideSize = sideSize - newSize;
```

Comment:
seems like we don't need to change sideSize because it will be used to calculate ratio like: (main panel size) / (split size)
so we can apply this ratio as % width or height and get correct size